packages/models/src/serialize.js

```diff
@@ -92,8 +92,16 @@ export function deserializeFilter(filterState) {
   const filter = new AppMapFilter();
   if (!filterState) return filter;
 
-  if ('rootObjects' in filterState) {
-    filter.declutter.rootObjects = filterState.rootObjects;
+  for (const property in filterState) {
+    if (
+      Object.prototype.hasOwnProperty.call(filterState, property) &&
+      filterState[property] === undefined
+    )
+      delete filterState[property];
+  }
+
+  if ('rootObjects' in filterState && filterState.rootObjects !== false) {
+    filter.rootObjects = filterState.rootObjects;
```

Comment:
Changing it from `filter.declutter.rootObjects` to `filter.rootObjects` is going to break some other code that's looking for `rootObjects` in the `declutter` property. It looks like that's why the build failed. Should we just leave it as `filter.declutter.rootObjects`?

Reply:
Oh true, I had assumed that the test assertions were reflective of actual code, but that wasn't a good assumption.